File: Examples/CSharp/Convert/GetConvertTradeHistory_Example.cs
namespace Binance.Spot.ConvertExamples
{
    using System.Net;
    using System.Net.Http;
    using System.Threading.Tasks;
    using Binance.Common;
    using Binance.Spot;
    using Binance.Spot.Models;
    using Microsoft.Extensions.Logging;

    public class GetConvertTradeHistory_Example
    {
        public static async Task Main(string[] args)
        {
            using var loggerFactory = LoggerFactory.Create(builder =>
            {
                builder.AddConsole();
            });
            ILogger logger = loggerFactory.CreateLogger<GetConvertTradeHistory_Example>();

            HttpMessageHandler loggingHandler = new BinanceLoggingHandler(logger: logger);
            HttpClient httpClient = new HttpClient(handler: loggingHandler);

            var convert = new Convert(httpClient);

            var result = await convert.GetConvertTradeHistory(1639646747000, 1642325147000);
        }
    }
}

File: Examples/FSharp/Convert/GetConvertTradeHistory_Example.fs
open System.Net
open System.Net.Http
open System.Threading.Tasks
open Microsoft.Extensions.Logging
open Binance.Common
open Binance.Spot
open Binance.Spot.Models

[<EntryPoint>]
let main argv =
    let loggerFactory = LoggerFactory.Create(fun (builder:ILoggingBuilder) -> 
        builder.AddConsole() |> ignore
    )
    let logger = loggerFactory.CreateLogger()

    let loggingHandler = new BinanceLoggingHandler(logger)
    let httpClient = new HttpClient(loggingHandler)

    let apiKey = "api-key";
    let apiSecret = "api-secret";
    
    let convert = new Convert(httpClient, apiKey = apiKey, apiSecret = apiSecret)
    
    let result = convert.GetConvertTradeHistory(1563189166000L, 1563282766000L) |> Async.AwaitTask |> Async.RunSynchronously
    
    0
